rp2040-hal/src/timer.rs

```diff
@@ -65,47 +77,30 @@ impl Timer {
             next_end: None,
         }
     }
-
     /// Retrieve a reference to alarm 0. Will only return a value the first time this is called
     pub fn alarm_0(&mut self) -> Option<Alarm0> {
-        if self.alarms[0] {
-            self.alarms[0] = false;
-            Some(Alarm0(PhantomData))
-        } else {
-            None
-        }
+        take_alarm(1 << 0).then_some(Alarm0(PhantomData))
     }
 
     /// Retrieve a reference to alarm 1. Will only return a value the first time this is called
     pub fn alarm_1(&mut self) -> Option<Alarm1> {
-        if self.alarms[1] {
-            self.alarms[1] = false;
-            Some(Alarm1(PhantomData))
-        } else {
-            None
-        }
+        take_alarm(1 << 1).then_some(Alarm1(PhantomData))
     }
 
     /// Retrieve a reference to alarm 2. Will only return a value the first time this is called
     pub fn alarm_2(&mut self) -> Option<Alarm2> {
-        if self.alarms[2] {
-            self.alarms[2] = false;
-            Some(Alarm2(PhantomData))
-        } else {
-            None
-        }
+        take_alarm(1 << 2).then_some(Alarm2(PhantomData))
     }
 
     /// Retrieve a reference to alarm 3. Will only return a value the first time this is called
     pub fn alarm_3(&mut self) -> Option<Alarm3> {
-        if self.alarms[3] {
-            self.alarms[3] = false;
-            Some(Alarm3(PhantomData))
-        } else {
-            None
-        }
+        take_alarm(1 << 3).then_some(Alarm3(PhantomData))
     }
 }
+// safety: there can only be only 1 instance of this alarm so no shared mutable access
+unsafe impl Send for Timer {}
```

Comment:
Also already implemented, https://docs.rs/rp2040-hal/latest/rp2040_hal/timer/struct.Timer.html#impl-Send-for-Timer
(but not Sync)

Reply:
hmm indeed, I got a bit heavy handed :P